lucene/test-framework/src/java/org/apache/lucene/codecs/asserting/AssertingDocValuesFormat.java

```diff
@@ -285,7 +286,9 @@ public SortedSetDocValues getSortedSet(FieldInfo field) throws IOException {
       assert field.getDocValuesType() == DocValuesType.SORTED_SET;
       SortedSetDocValues values = in.getSortedSet(field);
       assert values != null;
-      return new AssertingLeafReader.AssertingSortedSetDocValues(values, maxDoc);
+      final SortedDocValues singleton = DocValues.unwrapSingleton(values);
+      return singleton==null ? new AssertingLeafReader.AssertingSortedSetDocValues(values, maxDoc) :
```

Comment:
I think it's worth to span this to NumericsSet and also to other usages AssertingDV. Now, these DVs aren't handled by `DocValues.unwrapSingleton()`.
@romseygeek , isn't it worth to commit separately?